src/qlipperpreferencesdialog.cpp

```diff
@@ -73,6 +76,22 @@ QlipperPreferencesDialog::QlipperPreferencesDialog(QWidget *parent) :
         stickyRemoveButton->setEnabled(false);
     }
 
+    // Setup icon image machinery.
+    // Setup menu & actions for icon selection.
+    QMenu *iconMenu = new QMenu(tr("Icon selection"), this);
+    QAction *actLoadFromFile = new QAction(tr("Load icon from file..."), this);
+    QAction *actUseDefault = new QAction(QIcon(DEFAULT_ICON_PATH), tr("Use default icon"), this);
+
+    iconMenu->addAction(actLoadFromFile);
+    iconMenu->addAction(actUseDefault);
+
+    temporarilyRembemberNewTrayIcon(s->getPathToIcon());
+
+    buttonIconImage->setMenu(iconMenu);
+
+    connect(actLoadFromFile, SIGNAL(triggered()), this, SLOT(selectIconFromFile()));
```

Comment:
> Should it be done in separate commits outside this PR?

Just change your new `connect`(s) here and it would be great, if you could make PR for all other usages.